portal/models/questionnaire_bank.py

```diff
@@ -278,41 +283,40 @@ def most_current_qb(user, as_of_date=None):
 
         baseline = QuestionnaireBank.qbs_for_user(user, 'baseline')
         if not baseline:
-            return (None, None)
+            return QBD(None, None, None, None)
         trigger_date = baseline[0].trigger_date(user)
         if not trigger_date:
-            return (None, None)
+            return QBD(None, None, None, None)
 
         # Iterate over users QBs looking for current
-        last_found = baseline[0]
-        last_found_ic = None
+        last_found = QBD(relative_start=None, iteration=None, recur=None,
+                         questionnaire_bank=baseline[0])
         for classification in classification_types:
             if classification == 'indefinite':
                 continue
             for qb in QuestionnaireBank.qbs_for_user(user, classification):
-                relative_start, ic = qb.calculated_start(trigger_date,
-                                                         as_of_date)
-                if relative_start is None:
+                qbd = qb.calculated_start(trigger_date, as_of_date)
+                if qbd.relative_start is None:
                     # indicates QB hasn't started yet, continue
                     continue
                 expiry = qb.calculated_expiry(trigger_date)
-                last_found = qb
-                last_found_ic = ic
+                last_found = qbd._replace(questionnaire_bank=qb)
```

Comment:
seems like the iteration should be captured in the `qbd._replace()` as well?

Reply:
It should already be captured (because `qbd` is returned from `calculated_start()`, which contains the current iteration)

Reply:
ah!  thanks.